mods/creative/inventory.lua

```diff
@@ -0,0 +1,162 @@
+					def.description:lower():find(inv.filter, 1, true)) then
+			creative_list[#creative_list+1] = name
+		end
+	end
+
+	table.sort(creative_list)
+	player_inv:set_size("main", #creative_list)
+	player_inv:set_list("main", creative_list)
+	inv.size = #creative_list
+end
+
+-- Create the trash field
+local trash = minetest.create_detached_inventory("creative_trash", {
+	-- Allow the stack to be placed and remove it in on_put()
+	-- This allows the creative inventory to restore the stack
+	allow_put = function(inv, listname, index, stack, player)
+		return stack:get_count()
+	end,
+	on_put = function(inv, listname)
+		inv:set_list(listname, {})
+	end,
+})
+trash:set_size("main", 1)
+
+creative.formspec_add = ""
+
+function creative.register_tab(name, title, items)
+	sfinv.register_page("creative:" .. name, {
+		title = title,
+		get = function(self, player, context)
+			local player_name = player:get_player_name()
+			creative.update_creative_inventory(player_name, items)
+			local inv = player_inventory[player_name]
+			local start_i = inv.start_i or 0
+			local pagenum = math.floor(start_i / (3*8) + 1)
+			local pagemax = math.ceil(inv.size / (3*8))
+			return sfinv.make_formspec(player, context,
+				"label[6.2,3.35;" .. minetest.colorize("#FFFF00", tostring(pagenum)) .. " / " .. tostring(pagemax) .. "]" ..
+				[[
+					listcolors[#00000069;#5A5A5A;#141318;#30434C;#FFF]
+					list[current_player;main;0,4.7;8,1;]
+					list[current_player;main;0,5.85;8,3;8]
+					list[detached:creative_trash;main;4,3.3;1,1;]
+					listring[]
+					button[5.4,3.2;0.8,0.9;creative_prev;<]
+					button[7.25,3.2;0.8,0.9;creative_next;>]
+					button[2.1,3.4;0.8,0.5;creative_search;?]
+					button[2.75,3.4;0.8,0.5;creative_clear;X]
+					tooltip[creative_search;Search]
+					tooltip[creative_clear;Reset]
+					listring[current_player;main]
+				]] ..
+				"field[0.3,3.5;2.2,1;creative_filter;;" .. minetest.formspec_escape(inv.filter) .. ";false]" ..
+				"listring[detached:creative_" .. player_name .. ";main]" ..
+				"list[detached:creative_" .. player_name .. ";main;0,0;8,3;" .. tostring(start_i) .. "]" ..
+				default.get_hotbar_bg(0,4.7) ..
+				default.gui_bg .. default.gui_bg_img .. default.gui_slots
+				.. creative.formspec_add, false)
+		end,
+		on_enter = function(self, player, context)
+			local player_name = player:get_player_name()
+			local inv = player_inventory[player_name]
+			if inv then
+				inv.start_i = 0
+			end
+		end,
+		on_player_receive_fields = function(self, player, context, fields)
+			local player_name = player:get_player_name()
+			local inv = player_inventory[player_name]
+
+			if fields.creative_clear then
+				inv.start_i = 0
+				inv.filter = ""
+				creative.update_creative_inventory(player_name, items)
+				sfinv.set(player, context)
+			elseif fields.creative_search or
+					fields.key_enter_field == "creative_filter" then
+				inv.start_i = 0
+				inv.filter = fields.creative_filter:lower()
```

Comment:
I'm not sure how to reproduce: do I need an old version of MT? If so, at which commit? What do you mean by:

> before the formspec changes